packages/atomic/src/components/search/result-template-components/atomic-quickview-modal/atomic-quickview-modal.tsx

```diff
@@ -75,17 +57,34 @@ export class AtomicQuickviewModal implements InitializableComponent {
 
   private renderBody() {
     return (
-      <div
-        slot="body"
-        class="grid grid-cols-[min-content_auto]"
-        style={{height: '90vh'}}
-      >
-        <div class="h-full">{this.renderSidebar()}</div>
-        <div class="overflow-auto">
+      <div slot="body" class="grid grid-cols-[min-content_auto] h-full">
+        <div class="h-full">
+          <QuickviewSidebar
+            words={this.words}
+            i18n={this.bindings.i18n}
+            highlightKeywords={this.highlightKeywords}
+            onHighlightKeywords={(highlight) =>
+              (this.highlightKeywords = highlight)
+            }
+            minimized={this.minimizeSidebar}
+            onMinimize={(minimize) => (this.minimizeSidebar = minimize)}
+          />
+        </div>
+        <div class="overflow-auto relative">
           <QuickviewIframe
+            result={this.result}
             content={this.content}
-            onSetIframeRef={(ref) => (this.iframeRef = ref)}
+            onSetIframeRef={(ref) => {
+              this.iframeRef = ref;
+              setTimeout(() => {
```

Comment:
Is this because the height of the elements/iframe aren't correct until we wait a bit?

If so, I would prefer if we somehow named this delay. E.g.:
```js
this.iframeRef = ref;
await this.waitForIFrameHeightCalculations(); // Just returns a `new Promise<void>(resolve => setTimeout(resolve));`
this.words = getWordsHighlights(
  this.termsToHighlight,
  this.iframeRef
);
```

Reply:
Alternatively, we could abstract away the iframe when it exits `QuickviewIframe`. `QuickviewIframe` could return the bare minimum (e.g., `document` and/or `documentWriter`) instead of the iframe, and it could also take care of waiting for height calculations to be done before calling this callback prop.

E.g.:
```tsx
<QuickviewIframe
  result={this.result}
  content={this.content}
  onContentRendered={({ renderedContent, writer }) => {
    this.renderedContent = renderedContent;
    this.writer = writer;
    this.words = getWordsHighlights(
      this.termsToHighlight,
      this.writer
    );
  }}
/>
```

Reply:
It's because when you `document.write` and `document.close` on a document (iframe included), it's not synchronous. 

It's written at the end of the current call stack.

So before you can use `querySelector` on it and have it contains the right content to calculate the word map, we need to push it back.

I can add a name for the wait 👍